src/app/ReadClient.cpp

```diff
@@ -391,12 +391,18 @@ CHIP_ERROR ReadClient::ProcessReportData(System::PacketBufferHandle && aPayload)
         TLV::TLVReader attributeReportIBsReader;
         attributeReportIBs.GetReader(&attributeReportIBsReader);
 
-        mpCallback->OnReportBegin(this);
+        if (IsInitialReport())
```

Comment:
this seems to be problematic, currently mInitialReport  in readClient is set to true in init and shutdown, then after receiving the initial priming report, it is set to false. During subscription, for every beginning report and end report, it needs to call OnReportBegin and OnReportEnd, then recover the mInitialReport to true for next round...

before this PR, IsInitialReport() is used to get the subscriptionid, I suggest to have IsPrimingReport for this purpose, then make isInitialReport for chunking report begin purpose.

Reply:
https://github.com/project-chip/connectedhomeip/issues/12128

Reply:
Also suggest to add unit test to check OnReportBegin and OnReportEnd